### PR TITLE
React UI: Fix "<path-prefix>/new" -> "<path-prefix>/new/" redirect

### PR DESCRIPTION
**Pull request by juliusv**
Fixes https://github.com/prometheus/prometheus/issues/6163

Signed-off-by: Julius Volz <julius.volz@gmail.com>

### CODE REVIEW

**Comment by juliusv** [comment]
@roidelapluie

File: web/web.go [hunk 1]
```diff
@@ -347,6 +347,20 @@ func New(logger log.Logger, o *Options) *Handler {
...
 	})
 
+	// Make sure that "<path-prefix>/new" is redirected to "<path-prefix>/new/" and
+	// not just the naked "/new/", which would be the default behavior of the router
+	// with the "RedirectTrailingSlash" option (https://godoc.org/github.com/julienschmidt/httprouter#Router.RedirectTrailingSlash),
+	// and which breaks users with a --web.route-prefix that deviates from the path derived
+	// from the external URL.
+	router.Get("/new", func(w http.ResponseWriter, r *http.Request) {
+		p := path.Join(o.ExternalURL.Path, "/new")
+		if p != "/" {
```
Comment (roidelapluie) [inline web/web.go]: p will always be different than / because we add /new at the end

File: web/web.go [hunk 2]
```diff
@@ -347,6 +347,20 @@ func New(logger log.Logger, o *Options) *Handler {
 		fs.ServeHTTP(w, r)
 	})
 
+	// Make sure that "<path-prefix>/new" is redirected to "<path-prefix>/new/" and
+	// not just the naked "/new/", which would be the default behavior of the router
+	// with the "RedirectTrailingSlash" option (https://godoc.org/github.com/julienschmidt/httprouter#Router.RedirectTrailingSlash),
+	// and which breaks users with a --web.route-prefix that deviates from the path derived
+	// from the external URL.
+	router.Get("/new", func(w http.ResponseWriter, r *http.Request) {
+		p := path.Join(o.ExternalURL.Path, "/new")
```
Comment (roidelapluie) [inline web/web.go]: should be join with just "new" ? then next line if == "new" then "/" + path

**Comment by juliusv** [comment]
Hurrdurr, I should really turn on my brain when doing quick PRs :laughing: Update in 3... 2... 1...

**Comment by juliusv** [comment]
Look again :)

**Comment by juliusv** [comment]
Hm wait...

**Comment by juliusv** [comment]
Ah no, should be all ok now.

**Comment by roidelapluie** [comment]
I tested multiple variations of `--web.route-prefix` and `--web.external-url` and it works:
```
#!/bin/bash
f=0

./prometheus &>/dev/null &
sleep 1;
curl -s 127.0.0.1:9090/new | grep '"/new/"' || f=1
PID=$!
kill $PID


./prometheus --web.route-prefix /UNIQ &>/dev/null &
sleep 1;
curl -s 127.0.0.1:9090/UNIQ/new | grep '"/UNIQ/new/"' || f=1
PID=$!
kill $PID

./prometheus --web.route-prefix /REQ --web.external-url http://127.0.0.1:9090/RESP  &>/dev/null &
sleep 1;
curl -s 127.0.0.1:9090/REQ/new | grep '"/RESP/new/"' || f=1
PID=$!
kill $PID

./prometheus --web.external-url http://127.0.0.1:9090/UNIQ  &>/dev/null &
sleep 1;
curl -s 127.0.0.1:9090/UNIQ/new | grep '"/UNIQ/new/"' || f=1
PID=$!
kill $PID

if [[ $f -eq 0 ]]
then
    echo All tests OK
else
    echo Failure
fi
```

**Comment by juliusv** [comment]
Thanks for testing!

**Pull request merged**